docs/releasing/publishing-a-pre-release.md

```diff
@@ -0,0 +1,125 @@
+# Before you publish a pre-release of GOV.UK Frontend
+
+Decide the kind of pre-release you're publishing. All pre-releases are public, but we distinguish:
+
+- `internal` pre-releases for internal use
+- `beta` pre-releases aimed at external users
+
+We'll use the kind of pre-release as [the pre-release identifier](https://docs.npmjs.com/cli/v8/commands/npm-version#preid) in the package version.
+
+## If you're publishing a beta pre-release
+
+Besides publishing the code itself, beta releases will likely involve
+documentation updates linked to the code, as well as communications, similarly to an actual release.
+
+Review the docs for [what to do before publishing a release](/docs/releasing/before-publishing-a-release.md) to assess which steps you need to follow for your specific pre-release and ensure you are prepared to publish.
+
+See the [documentation on support branches](https://govuk-design-system-team-docs.netlify.app/how-we-work/version-control/support-branches.html#support-branches) if you need to publish a new pre-release of previous major versions of GOV.UK Frontend.
+
+# Publish a new version of GOV.UK Frontend
+
+Developers should pair on pre-releases. When remote working, it can be useful to be on a call together.
+
+1. Check out the **main** branch and pull the latest changes.
+
+2. Run `nvm use` to make sure you're using the right version of Node.js and npm.
+
+3. Run `npm ci` to make sure you have the exact dependencies installed.
+
+4. Determine the pre-release identifier
+
+   - Use `internal` for internal pre-releases
+   - Use `beta` for beta pre-releases
+
+5. Determine the pre-release version type
+
+   Given the `beta` pre-release identifier:
```

Comment:
This might indicate that the following options are only for `beta` pre-releases

With enough "for beta, do this" or "for internal, do this" variables, should we split the two paths out more?